src/libraries/Microsoft.Extensions.Configuration/src/ChainedConfigurationProvider.cs

```diff
@@ -29,7 +29,9 @@ public ChainedConfigurationProvider(ChainedConfigurationSource source)
             }
             if (source.Configuration == null)
             {
+#pragma warning disable CA2208 // Instantiate argument exceptions correctly
```

Comment:
Seems like the analyzer shouldn't have warned here.  The expression passed to `nameof` includes an argument, so while the nameof text won't evaluate to include it, the developer is obviously validating something relating to that argument and making a decision to dot off of it.

Reply:
We had discussed about should we if we allow `nameof(parameter.Property)` or `"parameter.Property"` and concluded it is invalid, we should pass an argument and the `Property` in question should be passed to the exception message. CC @terrajobst @bartonjs 
That is not working well for this case as there is no message provided …

Reply:
If we concluded it's invalid, then why are we doing it?

Personally, I don't think it's invalid, though I also think the current form employed here isn't great, since it's not clear when someone gets the string "Configuration" that it's actually referring to a property on "source".  My preference in a case like this would either be for the string to be "source.Configuration" (e.g. `nameof(source) + "." + nameof(source.Configuration)`) and for the analyzer to not warn on any nameof that includes the target parameter as part of the nameof expression, or for the argument to just be `nameof(source)` and update the message to include `Configuration`.

Reply:
> If we concluded it's invalid, then why are we doing it

Sorry i meant invalid for paramName argument of an argument exception. That time we were discussing the case with error message and decided to pass the Property to the message instead

But in this case there is no message so for now I just suppressed

Reply:
I am thinking to update analyzer to allow `nameof(source) + "." + nameof(source.Configuration)` or `"source.Configuration"` in case there is no exception message to place the property name CC @bartonjs @terrajobst

Reply:
@terrajobst didn't approve allowing `nameof(source) + "." + nameof(source.Configuration) or "source.Configuration"` because: "ANE is not right thing to throw here as arguments was not null",  instead he is suggesting to throw ArgumentException with a message something like "ChainedConfigurationSource needs to have a source". 

No exception documented for this API, so am gonna apply this suggestion